system/security/shared/caching.cpp

```diff
@@ -353,7 +353,7 @@ bool CPermissionsCache::lookup(ISecUser& sec_user)
 
         time_t now;
         time(&now);
-        if(user->getTimestamp() < (now - m_cacheTimeout)  && 0==sec_user.credentials().getSessionToken())//don't delete session based users
+        if(0==sec_user.credentials().getSessionToken()  &&  (user->getTimestamp() < (now - m_cacheTimeoutInSeconds)))//don't delete session based users
```

Comment:
not directly connected to this change, but the code above should be expressed as

     (now - user->getTimestamp()) > m_cacheTimeoutInSeconds.

otherwise if  user->getTimeStamp() >= (MAX_LONG - m_cacheTimeoutInSeconds) it will never time out.